packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -1,11 +1,60 @@
 import { useEffect, useState } from 'react'
+import { isAddress } from 'ethers/lib/utils'
+import { Provider } from '@ethersproject/providers'
 import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/outline'
 
 import { useAppState } from '../../state'
 import { useAccountType } from '../../hooks/useAccountType'
+import { addressIsSmartContract } from '../../util/AddressUtils'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
 
-export enum AdvancedSettingsErrors {
-  INVALID_ADDRESS = 'The destination address is not valid.'
+export enum DestinationAddressErrors {
+  INVALID_ADDRESS = 'The destination address is not a valid wallet address.',
+  REQUIRED_ADDRESS = 'The destination address is required.'
+}
+
+enum DestinationAddressWarnings {
+  CONTRACT_ADDRESS = 'The destination address is a contract address. Please make sure it is a valid wallet address.'
+}
+
+export function getDestinationAddressError({
+  destinationAddress,
+  isEOA
+}: {
+  destinationAddress?: string
+  isEOA: boolean
+}): DestinationAddressErrors | null {
+  if (!destinationAddress) {
+    if (isEOA) {
+      return null
+    }
+    // destination address required for contract wallets
+    return DestinationAddressErrors.REQUIRED_ADDRESS
+  }
+  if (!isAddress(destinationAddress)) {
+    return DestinationAddressErrors.INVALID_ADDRESS
+  }
+  return null
+}
+
+async function getDestinationAddressWarning({
+  destinationAddress,
+  isEOA,
+  providerTo
+}: {
+  destinationAddress: string
+  isEOA: boolean
+  providerTo: Provider
+}) {
+  // checks if trying to send to a contract address
+  // only check for EOA, contract wallets will often send to another contract wallet
+  if (!isEOA || !isAddress(destinationAddress)) {
+    return null
+  }
+  if (await addressIsSmartContract(destinationAddress, providerTo)) {
+    return DestinationAddressWarnings.CONTRACT_ADDRESS
+  }
+  return null
```

Comment:
I might be missing something but this can be simplified?

```suggestion
  const destinationIsSmartContract = await addressIsSmartContract(destinationAddress, providerTo)

  if (isEOA && destinationIsSmartContract) {
    return DestinationAddressWarnings.CONTRACT_ADDRESS
  }
  
  return null
```

Reply:
It can, updated